### PR TITLE
Human Task Editor for Eclipse Using Web UI Platform

### PR DESCRIPTION
**Pull request by Drifftr**

### CODE REVIEW

File: features/org.wso2.developerstudio.eclipse.bps.feature/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (amalhub) [inline features/org.wso2.developerstudio.eclipse.bps.feature/pom.xml]: Please add the WSO2 copyright comments to all the files.

File: features/org.wso2.developerstudio.eclipse.bps.feature/feature.common.properties [hunk 2]
```diff
@@ -0,0 +1,217 @@
+################################################################################
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (amalhub) [inline features/org.wso2.developerstudio.eclipse.bps.feature/feature.common.properties]: Please update the copyright year to the latest in comments.

File: installed-distribution/developer-studio-bps-eclipse-jee-luna-linux-gtk-x86_64/pom.xml [hunk 3]
```diff
@@ -0,0 +1,180 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~ Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (amalhub) [inline installed-distribution/developer-studio-bps-eclipse-jee-luna-linux-gtk-x86_64/pom.xml]: Copyright year is missing.

File: installed-distribution/pom.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            <dependency>
+                <groupId>org.wso2.developerstudio</groupId>
+                <artifactId>wso2-developerstudio-bps-toolkit</artifactId>
+                <version>4.0.0-SNAPSHOT</version>
+                <type>zip</type>
+            </dependency>
+            <dependency>
+                <groupId>org.wso2.carbon</groupId>
+                <artifactId>wso2-p2-agent</artifactId>
+                <version>3.1.0</version>
```
Comment (amalhub) [inline installed-distribution/pom.xml]: It's better to define the versions as properties in the root pom and reference.

File: plugins/org.wso2.developerstudio.humantaskeditor/.gitignore [hunk 5]
```diff
@@ -0,0 +1,70 @@
+/bin/
+/bin/
+.classpath
+.project
+.settings
+/target/
+
+/bin/
```
Comment (amalhub) [inline plugins/org.wso2.developerstudio.humantaskeditor/.gitignore]: The bin folder has been added to the .gitignore file multiple times. Better to remove unnecessary lines.

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js [hunk 6]
```diff
@@ -0,0 +1,1108 @@
...
+
+    // Create Task WSDL
+    $.get('resources/dummy.wsdl', function(data) {
+        try {
+            wsdlDom = marshalEditorTextContent(data);
+            generateInputWSDL(wsdlDom, currentTaskName);
+            saveWSDL(wsdlDom, currentTaskName + "Task");
+
+        } catch (err) {
+            alert(err);
```
Comment (amalhub) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js]: Better if there's a way to show the error message in a well-formed message dialog box than using the default alert function.

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/index.html [hunk 7]
```diff
@@ -0,0 +1,508 @@
...
+<head>
+    <META http-equiv="Content-Type" content="text/html; charset=utf-8">
+    <title>Untitled Document</title>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+    <link href="css/main.css" rel="stylesheet">
+    <link href="css/bootstrap.min.css" rel="stylesheet">
+    <link href="css/jquery-ui.css" rel="stylesheet">
+    <script src="js/jquery-2.1.4.min.js"></script>
+    <script type="text/javascript" src="js/jquery-ui.js"></script>
+    <script src="js/bootstrap.min.js"></script>
```
Comment (amalhub) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/index.html]: Better to load the JS scripts at the bottom of the page. This enables to load the page lot faster.

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/js/WSDLOperations.js [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+    // get Definitions node and Schema node from dummy wsdl dom
+    wsdlSchema = wsdlDom.getElementsByTagName("schema")[0];
+    wsdlDefinitions = wsdlDom.getElementsByTagName("definitions")[0];
+
+    // create Data type elements
+    newComplexType = wsdlDom.createElementNS(
+        "http://www.w3.org/2001/XMLSchema", "xsd:complexType");
+    newComplexType.setAttribute("name", currentTask + "DataType"); // Data type
+    // Name
+    newSequence = wsdlDom.createElementNS("http://www.w3.org/2001/XMLSchema",
```
Comment (amalhub) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/js/WSDLOperations.js]: "http://www.w3.org/2001/XMLSchema"
Following string has been used in many places over the script. It's better to create a constant variable and use it where necessary.

File: features/org.wso2.developerstudio.eclipse.bps.feature/.project [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hasithaa) [inline features/org.wso2.developerstudio.eclipse.bps.feature/.project]: Missing License Header.

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/js/WSDLOperations.js [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+    newMessage.setAttribute("name", currentTask + "Request");
+    newPart = wsdlDom.createElementNS("http://schemas.xmlsoap.org/wsdl/",
+        "wsdl:part");
+    newPart.setAttribute("name", newMessage.getAttribute("name"));
+    newPart.setAttribute("element", "tschema:" + newElement.getAttribute("name")); // Data binding
+    newMessage.appendChild(newPart);
+    wsdlDefinitions.appendChild(newMessage);
+
+    // create portType element
+    newPortType = wsdlDom.createElementNS("http://schemas.xmlsoap.org/wsdl/",
```
Comment (amalhub) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/js/WSDLOperations.js]: "http://schemas.xmlsoap.org/wsdl/"
Following string has been used in many places over the script. It's better to create a constant variable and use it where necessary.

File: features/org.wso2.developerstudio.eclipse.bps.feature/build.properties [hunk 11]
```diff
@@ -0,0 +1,7 @@
+src.includes = feature.xml,\
```
Comment (hasithaa) [inline features/org.wso2.developerstudio.eclipse.bps.feature/build.properties]: Missing License Header.

File: features/org.wso2.developerstudio.eclipse.bps.feature/feature.xml [hunk 12]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hasithaa) [inline features/org.wso2.developerstudio.eclipse.bps.feature/feature.xml]: Missing License Header. Check this for all new and existing files. If you did a modification to existing file, update current year in license text.

File: features/pom.xml [hunk 13]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (hasithaa) [inline features/pom.xml]: Missing Year.

File: plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/editors/HumanTaskMultiPageEditor.java [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+     * editor font used in page 2.
+     */
+    void createPage0() {
+
+        humanTaskUIEditor = new HumanTaskUIEditor();
+
+        int index;
+        try {
+            index = addPage(humanTaskUIEditor, getEditorInput());
+            setPageText(index, "GUI Editor");
```
Comment (amalhub) [inline plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/editors/HumanTaskMultiPageEditor.java]: Better to define these strings as constants in a separate class and refer.

File: plugins/pom.xml [hunk 15]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (amalhub) [inline plugins/pom.xml]: Missing license header.

File: pom.xml [hunk 16]
```diff
@@ -0,0 +1,252 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (amalhub) [inline pom.xml]: Missing year in license header

File: repository/composite/category.xml [hunk 17]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (amalhub) [inline repository/composite/category.xml]: Missing license header.

File: repository/composite/pom.xml [hunk 18]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+		~ Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved. ~ ~
```
Comment (amalhub) [inline repository/composite/pom.xml]: License header year is missing.

File: repository/main/category.xml [hunk 19]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (amalhub) [inline repository/main/category.xml]: Missing license header

File: repository/main/pom.xml [hunk 20]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+		~ Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved. ~ ~
```
Comment (amalhub) [inline repository/main/pom.xml]: License header year is missing.

File: repository/pom.xml [hunk 21]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+		~ Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved. ~ ~
```
Comment (amalhub) [inline repository/pom.xml]: License header year is missing.

File: features/feature.common.properties [hunk 22]
```diff
@@ -0,0 +1,217 @@
+################################################################################
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (amalhub) [inline features/feature.common.properties]: License header year is not the latest.

File: installed-distribution/developer-studio-bps-eclipse-jee-luna-linux-gtk-x86_64/src/assembly/dist.xml [hunk 23]
```diff
@@ -0,0 +1,34 @@
+<!--
+ ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (amalhub) [inline installed-distribution/developer-studio-bps-eclipse-jee-luna-linux-gtk-x86_64/src/assembly/dist.xml]: License header year needs to be latest.

File: installed-distribution/developer-studio-bps-eclipse-jee-luna-linux-gtk/pom.xml [hunk 24]
```diff
@@ -0,0 +1,180 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~ Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (amalhub) [inline installed-distribution/developer-studio-bps-eclipse-jee-luna-linux-gtk/pom.xml]: License header year is missing.

File: feature-updater/pom.xml [hunk 25]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+~
```
Comment (nandika) [inline feature-updater/pom.xml]: Copyright year missing in the license header

File: feature-updater/pom.xml [hunk 26]
```diff
@@ -0,0 +1,93 @@
...
+~ See the License for the specific language governing permissions and
+~ limitations under the License.
+-->
+<project
+    xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wso2.developerstudio</groupId>
+    <artifactId>wso2-developerstudio-esb-updates-publisher</artifactId>
+    <version>4.0.0-SNAPSHOT</version>
```
Comment (nandika) [inline feature-updater/pom.xml]: Need to correct the artifact id

File: plugins/org.wso2.developerstudio.eclipse.embedded.tomcat/src/org/wso2/developerstudio/eclipse/embedded/tomcat/EmbeddedTomcatPlugin.java [hunk 27]
```diff
@@ -0,0 +1,120 @@
...
+	public static final String PLUGIN_ID = "org.wso2.developerstudio.embedded.tomcat";//$NON-NLS-1$
+
+	private static ClassLoader bundleCtxClassLoader;
+	private static BundleContext context;
+	private static EmbeddedTomcatPlugin plugin;
+	private static IDeveloperStudioLog log = Logger.getLog(EmbeddedTomcatPlugin.PLUGIN_ID);
+
+	/** Tomcat instance */
+	private EmbeddedTomcatServer tomcatServer;
+
```
Comment (nandika) [inline plugins/org.wso2.developerstudio.eclipse.embedded.tomcat/src/org/wso2/developerstudio/eclipse/embedded/tomcat/EmbeddedTomcatPlugin.java]: Fix code formatting

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js [hunk 28]
```diff
@@ -0,0 +1,1049 @@
...
+ */
+function createFile(currentTaskName,state) { //createFile
+    // Create Task WSDL
+    var serviceURL = $('#' + currentTaskName + 'wrapper #taskCallbackServiceURL').val();
+    var operationName = $('#' + currentTaskName + 'wrapper #taskCallbackOperationName')
+        .val();
+    var serviceName = $('#' + currentTaskName + 'wrapper #taskCallbackServiceName').val();
+
+    var outputElements = $('#' + currentTaskName + 'wrapper #outputmappingTable tr');
+ $.get('resources/dummy.wsdl', function(data) {
```
Comment (nandika) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js]: code formatting isssue

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js [hunk 29]
```diff
@@ -0,0 +1,1053 @@
...
+                $('#' + taskDivName + " input[name="+peopleAssignmentName+"" + taskDivName + "][value="+peopleAssignmentName+"Literal]").prop("checked", true);
+                usersList = taskNode.getElementsByTagName("peopleAssignments")[0]
+                    .getElementsByTagName(peopleAssignmentName)[0]
+                    .getElementsByTagName("user");
+                groupsList = taskNode.getElementsByTagName("peopleAssignments")[0]
+                    .getElementsByTagName(peopleAssignmentName)[0]
+                    .getElementsByTagName("group");
+                for (var t = 0; t < usersList.length; t++) {
+                    if(usersList[t].childNodes.length!=0)
+                    $('#' + taskDivName + " #"+peopleAssignmentName+"LiteralUsers").val($('#' + taskDivName + " #"+peopleAssignmentName+"LiteralUsers").val() + ',' + usersList[t].childNodes[0].nodeValue);
```
Comment (himasha) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js]: Formatting issue

Comment (Drifftr) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js]: Fixed in https://github.com/wso2/devstudio-tooling-bps/pull/25

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js (hunk 28 above)
Comment (Drifftr) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js]: Fixed in https://github.com/wso2/devstudio-tooling-bps/pull/28/commits/2581f5ddab9d98e149879c6365b9582961715d55

**Pull request closed without merging**